### ISSUE TITLE
Utility of max_lr in OneCycleLR?

### ISSUE THREAD

**Issue by grudloff** [pytorch/pytorch#73305]
`OneCycleLR` uses the `max_lr` parameter to pass in the max learning rate for each param group, couldn't these values be inferred from the learning rates of each param group from the optimizer, as it is done with `CosineAnnealingLR`? What is the use case for requiring this "redundant" parameter?

This could be easily replaced internally as 
```python
self.max_lr = [param_group['lr'] for param_group in optimizer.param_groups]
```
And a transition without breaking backward compatibility could be easily achieved by setting a default `max_lr=None`, and in that case, the learning rates are inferred from the optimizer.

**Comment by george-qi** [comment]
Hi, thanks for the question! We use Github issues only for bug reports or feature requests. Please use the forum to ask questions: https://discuss.pytorch.org/

**Comment by grudloff** [comment]
Hello @george-qi! I guess this question is rather a feature request, I am questioning the design choice rather than asking a usage question that is more suitable for the forum. Maybe the title I choose is misleading?
